sdk/identity/azure-identity/src/test/java/com/azure/identity/util/TestUtils.java

```diff
@@ -135,6 +141,17 @@ public static Mono<AccessToken> getMockAccessToken(String accessToken, OffsetDat
         return Mono.just(new AccessToken(accessToken, expiresOn.plusMinutes(2).minus(tokenRefreshOffset)));
     }
 
+    /**
+     * Creates a {@link Configuration} with the specified {@link ConfigurationSource} as the only source of
+     * configurations.
+     *
+     * @param configurationSource The configuration source.
+     * @return A configuration used for testing.
+     */
+    public static Configuration createTestConfiguration(ConfigurationSource configurationSource) {
```

Comment:
nit: Can this optionally take a `null` and create a default `TestConfigurationSource` if so? All the callsites having to create one isn't great.